### PR TITLE
Shopping cart

### PR DESCRIPTION
**Pull request by Parashchuk**

### CODE REVIEW

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Your PR should not include changes from your previous HW - category.

File: src/main/java/bookstore/dto/cart/CartRequestAddDto.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package bookstore.dto.cart;
+
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.Positive;
+
+public record CartRequestAddDto(
+        @Positive
+        Long book_id,
+
+        @Min(value = 1)
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/dto/cart/CartRequestAddDto.java]: Why not ` @Positive `?

File: src/main/java/bookstore/dto/cart/CartRequestUpdateDto.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package bookstore.dto.cart;
+
+import jakarta.validation.constraints.Min;
+
+public record CartRequestUpdateDto(
+        @Min(value = 1)
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/dto/cart/CartRequestUpdateDto.java]: the same

File: src/main/java/bookstore/entity/cart/Cart.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+@Setter
+@ToString
+@Table(name = "carts")
+public class Cart {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @OneToOne
+    @JoinColumn(name = "user_id")
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/entity/cart/Cart.java]: nullable = false

File: src/main/java/bookstore/entity/cart/CartItem.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+@ToString
+@Table(name = "cart_items")
+@SQLDelete(sql = "UPDATE cart_items SET is_deleted = true WHERE id = ?")
+@Where(clause = "is_deleted = false")
+public class CartItem {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/entity/cart/CartItem.java]:
```suggestion
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
```

File: src/main/java/bookstore/entity/cart/CartItem.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    @JoinColumn(name = "cart_id")
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/entity/cart/CartItem.java]: nullable = false

File: src/main/java/bookstore/entity/cart/CartItem.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    private Cart cart;
+
+    @OneToOne
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/entity/cart/CartItem.java]: Wrong mapping 
@ManyToOne(fetch = FetchType.LAZY, optional = false)

Also, check Book entity and add cartItems

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's fix some comments

**Review by liliia-ponomarenko — APPROVED** [review]
Good job!

**Pull request merged**